packages/gatsby/src/redux/actions/public.js

```diff
@@ -616,6 +618,9 @@ const createNode = (
     node.internal = {}
   }
 
+  NODE_COUNTER++
+  node.internal.$counter = NODE_COUNTER
```

Comment:
Should this be added to the Joi schema validation? Why isn't it causing errors?

Reply:
Also do we need the $ prefix?

Reply:
It's not in loki because it's added after validation. I've moved it to be just "counter".